### PR TITLE
Exchange rate checker

### PR DESCRIPTION
**Pull request by faust403**

### CODE REVIEW

File: yarn.lock
Comment (foxpy) [inline yarn.lock:1]: Should this file really be here?

File: integration_tests/src/helpers/exchangeRate.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    return 1;
+  }
+  const exchangeRate: Decimal = exchangeRateNumerator.dividedBy(
+    exchangeRateDenominator,
+  );
+  return exchangeRate.toNumber();
+}
+
+export async function checkExchangeRate(testContext: any) {
+  const { coreContractClient, client, neutronRPCEndpoint } = testContext;
```
Comment (ratik) [inline integration_tests/src/helpers/exchangeRate.ts:125]: please use coreContractClient, puppeterContractClient, etc to query data. no need to connect once again

File: integration_tests/package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "test:gv": "vitest --run gv.test.ts --bail 1",
     "debug": "vitest --inspect-brk --bail 1 --threads false --run core.test.ts",
     "watch": "vitest",
+    "calc-exchange-rate": "ts-node ./src/helpers/exchangeRate.ts",
```
Comment (ratik) [inline integration_tests/package.json:21]: why?

**Review by ratik — CHANGES_REQUESTED** [review]

**Review by foxpy — APPROVED** [review]

File: integration_tests/src/helpers/exchangeRate.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+export async function calcExchangeRate(context: any): Promise<number> {
+  const queryClient = QueryClient.withExtensions(
+    await connectComet(context.neutronRPCEndpoint),
+    setupAuthExtension,
+    setupBankExtension,
+    setupStakingExtension,
+    setupTxExtension,
+  );
+  const RPC = createProtobufRpcClient(queryClient);
+  const queryService = new QueryClientImpl(RPC);
```
Comment (ratik) [inline integration_tests/src/helpers/exchangeRate.ts:24]: you can use neutronClient for this purpose I believe

**Pull request merged**